### PR TITLE
Clarify/fix some threading docs

### LINKED ISSUE #12214: Why ExecutionAndPublication and PublicationOnly have such names?
Could you please explain in the article, why ExecutionAndPublication and PublicationOnly have such names? The descriptions of modes do not mention publication and it is not clear how behavior is associated with the name of mode.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c082508c-af74-1d6a-a954-e70aca02e1a4
* Version Independent ID: 85c92c40-9a7a-cd0f-fa63-53c87cb5febe
* Content: [Lazy Initialization](https://docs.microsoft.com/en-us/dotnet/framework/performance/lazy-initialization?view=netframework-4.8)
* Content Source: [docs/framework/performance/lazy-initialization.md](https://github.com/dotnet/docs/blob/master/docs/framework/performance/lazy-initialization.md)
* Product: **dotnet-framework**
* GitHub Login: @rpetrusha
* Microsoft Alias: **ronpet**

### LINKED ISSUE #24318: Improve documentation of `volatile` keyword
Improve documentation of `volatile` keyword in `docs/csharp/language-reference/keywords/volatile.md` to clarify what it does and fix some inaccurate info.

-edit by @billwagner to add document block.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d226514d-c7cb-0d1b-b9dc-fb791399dc47
* Version Independent ID: aa5d1078-7f89-98ce-7fdc-107c9ed3eebc
* Content: [volatile - C# Reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/volatile)
* Content Source: [docs/csharp/language-reference/keywords/volatile.md](https://github.com/dotnet/docs/blob/main/docs/csharp/language-reference/keywords/volatile.md)
* Product: **dotnet-csharp**
* Technology: **csharp-language-reference**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #24566: Add docs for new config vars available to configure rate of thread injection for sync-over-async
Update https://github.com/dotnet/docs/blob/main/docs/core/run-time-config/threading.md to include new configs from https://github.com/dotnet/runtime/pull/53471

### PR DESCRIPTION
**Pull request by kouvel**
- Depends on https://github.com/dotnet/dotnet-api-docs/pull/7137
- Document new settings available to configure the rate of thread injection for sync-over-async. Fixes https://github.com/dotnet/docs/issues/24566.
- Clarify documentation of the `volatile` keyword. Fixes https://github.com/dotnet/docs/issues/24318.
- Add link to clarify some terms in `LazyThreadSafetyMode` field names. Fixes https://github.com/dotnet/docs/issues/12214.

### CODE REVIEW

File: docs/framework/performance/lazy-initialization.md [hunk 1]
```diff
@@ -79,6 +79,8 @@ ms.assetid: 56b4ae5c-4745-44ff-ad78-ffe4fcde6b9b
   
  As the table shows, specifying <xref:System.Threading.LazyThreadSafetyMode.ExecutionAndPublication?displayProperty=nameWithType> for the `mode` parameter is the same as specifying `true` for the `isThreadSafe` parameter, and specifying <xref:System.Threading.LazyThreadSafetyMode.None?displayProperty=nameWithType> is the same as specifying `false`.  
   
+ See <xref:System.Threading.LazyThreadSafetyMode> for more information about what `Execution` and `Publication` refer to.
```
Comment (gewarren) [inline docs/framework/performance/lazy-initialization.md:82]:
```suggestion
 For more information about what `Execution` and `Publication` refer to, see <xref:System.Threading.LazyThreadSafetyMode>.  
```

File: docs/csharp/language-reference/keywords/volatile.md [hunk 2]
```diff
@@ -11,10 +11,10 @@ ms.assetid: 78089bc7-7b38-4cfd-9e49-87ac036af009
 ---
 # volatile (C# Reference)
 
-The `volatile` keyword indicates that a field might be modified by multiple threads that are executing at the same time. The compiler, the runtime system, and even hardware may rearrange reads and writes to memory locations for performance reasons. Fields that are declared `volatile` are not subject to these optimizations. Adding the `volatile` modifier ensures that all threads will observe volatile writes performed by any other thread in the order in which they were performed. There is no guarantee of a single total ordering of volatile writes as seen from all threads of execution.
+The `volatile` keyword indicates that a field might be modified by multiple threads that are executing at the same time. The compiler, the runtime system, and even hardware may rearrange reads and writes to memory locations for performance reasons. Fields that are declared `volatile` are restricted from certain kinds of optimizations. There is no guarantee of a single total ordering of volatile writes as seen from all threads of execution. See the <xref:System.Threading.Volatile> class for more information.
```
Comment (gewarren) [inline docs/csharp/language-reference/keywords/volatile.md:14]:
```suggestion
The `volatile` keyword indicates that a field might be modified by multiple threads that are executing at the same time. The compiler, the runtime system, and even hardware may rearrange reads and writes to memory locations for performance reasons. Fields that are declared `volatile` are excluded from certain kinds of optimizations. There is no guarantee of a single total ordering of volatile writes as seen from all threads of execution. For more information, see the <xref:System.Threading.Volatile> class.
```

File: docs/core/run-time-config/threading.md [hunk 3]
```diff
@@ -94,6 +94,40 @@ Project file:
...
+| `System.Threading.ThreadPool.Blocking.ThreadsPerDelayStep_ProcCountFactor` | After the thread count based on `ThreadsToAddWithoutDelay` is reached, this value (after it is multiplied by the processor count) specifies after how many threads an additional DelayStepMs would be added to the delay before each new thread is created. |
+| `System.Threading.ThreadPool.Blocking.DelayStepMs` | After the thread count based on `ThreadsToAddWithoutDelay` is reached, this value specifies how much additional delay to add per `ThreadsPerDelayStep` threads, which would be applied before each new thread is created. |
+| `System.Threading.ThreadPool.Blocking.MaxDelayMs` | After the thread count based on `ThreadsToAddWithoutDelay` is reached, this value specifies the max delay to use before each new thread is created. |
+
+### How the configuration settings take effect
+
+- After the thread count based on `MinThreads` is reached, up to `ThreadsToAddWithoutDelay` additional threads may be created without a delay
```
Comment (gewarren) [inline docs/core/run-time-config/threading.md:110]:
```suggestion
- After the thread count based on `MinThreads` is reached, up to `ThreadsToAddWithoutDelay` additional threads may be created without a delay.
```

File: docs/core/run-time-config/threading.md [hunk 4]
```diff
@@ -94,6 +94,40 @@ Project file:
...
+- After that, before each additional thread is created, a delay is induced, starting with `DelayStepMs`
```
Comment (gewarren) [inline docs/core/run-time-config/threading.md:111]:
```suggestion
- After that, before each additional thread is created, a delay is induced, starting with `DelayStepMs`.
```

File: docs/core/run-time-config/threading.md [hunk 5]
```diff
@@ -94,6 +94,40 @@ Project file:
...
+- For every `ThreadsPerDelayStep` threads that are added with a delay, an additional `DelayStepMs` is added to the delay
```
Comment (gewarren) [inline docs/core/run-time-config/threading.md:112]:
```suggestion
- For every `ThreadsPerDelayStep` threads that are added with a delay, an additional `DelayStepMs` is added to the delay.
```

File: docs/core/run-time-config/threading.md [hunk 6]
```diff
@@ -94,6 +94,40 @@ Project file:
...
+- The delay may not exceed `MaxDelayMs`
```
Comment (gewarren) [inline docs/core/run-time-config/threading.md:113]:
```suggestion
- The delay may not exceed `MaxDelayMs`.
```

File: docs/core/run-time-config/threading.md [hunk 7]
```diff
@@ -94,6 +94,40 @@ Project file:
...
+- Delays are only induced before creating threads. If threads are already available, they would be released without delay to compensate for blocking work items.
+- Physical memory usage and limits are also used and beyond a threshold, the system switches to slower thread injection
```
Comment (gewarren) [inline docs/core/run-time-config/threading.md:115]:
```suggestion
- Physical memory usage and limits are also used and, beyond a threshold, the system switches to slower thread injection.
```

File: docs/core/run-time-config/threading.md [hunk 8]
```diff
@@ -94,6 +94,40 @@ Project file:
 </Project>
 ```
 
+## Thread injection in response to blocking work items
+
+In some cases the thread pool detects work items that block its threads and responds more quickly by injecting more threads to compensate. The following [runtime configuration](https://github.com/dotnet/docs/blob/main/docs/core/run-time-config/index.md) settings may be used to configure thread injection in response to blocking work items. Currently, these settings take effect only for work items that wait for another task to complete, such as in typical [sync-over-async](https://devblogs.microsoft.com/pfxteam/should-i-expose-synchronous-wrappers-for-asynchronous-methods/) cases.
```
Comment (gewarren) [inline docs/core/run-time-config/threading.md:99]:
```suggestion
In some cases, the thread pool detects work items that block its threads. To compensate, it injects more threads. You can use the following [runtime configuration](https://github.com/dotnet/docs/blob/main/docs/core/run-time-config/index.md) settings to configure thread injection in response to blocking work items. Currently, these settings take effect only for work items that wait for another task to complete, such as in typical [sync-over-async](https://devblogs.microsoft.com/pfxteam/should-i-expose-synchronous-wrappers-for-asynchronous-methods/) cases.
```

File: docs/core/run-time-config/threading.md [hunk 9]
```diff
@@ -94,6 +94,40 @@ Project file:
...
+
+| Setting name | Description |
```
Comment (gewarren) [inline docs/core/run-time-config/threading.md:101]:
```suggestion
| *runtimeconfig.json* setting name | Description |
```

**Review by gewarren — APPROVED** [review]

File: docs/core/run-time-config/threading.md [hunk 10]
```diff
@@ -94,6 +94,40 @@ Project file:
...
+| - | - |
+| `System.Threading.ThreadPool.Blocking.ThreadsToAddWithoutDelay_ProcCountFactor` | After the thread count based on `MinThreads` is reached, this value (after it is multiplied by the processor count) specifies how many additional threads may be created without a delay. |
+| `System.Threading.ThreadPool.Blocking.ThreadsPerDelayStep_ProcCountFactor` | After the thread count based on `ThreadsToAddWithoutDelay` is reached, this value (after it is multiplied by the processor count) specifies after how many threads an additional DelayStepMs would be added to the delay before each new thread is created. |
```
Comment (gewarren) [inline docs/core/run-time-config/threading.md:104]:
```suggestion
| `System.Threading.ThreadPool.Blocking.ThreadsPerDelayStep_ProcCountFactor` | After the thread count based on `ThreadsToAddWithoutDelay` is reached, this value (after it is multiplied by the processor count) specifies after how many threads an additional `DelayStepMs` would be added to the delay before each new thread is created. |
```

**Review by BillWagner — APPROVED** [review]
This LGTM @kouvel 

Once the related PR is merged, let's :shipit:

**Pull request merged**